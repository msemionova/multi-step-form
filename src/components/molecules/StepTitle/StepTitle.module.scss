.step {
  display: flex;
  align-items: center;
  gap: 1.4rem;

  h2, span {
    text-transform: uppercase;
  }

  h2 {
    font: var(--typography-step-title);
    color: var(--color-gray-background);
    letter-spacing: .1rem;
  }

  span {
    font: var(--typography-step-label);
    color: var(--color-gray-line);
  }

  &:not(:last-child) {
    margin-bottom: 2.4rem;
  }
}
