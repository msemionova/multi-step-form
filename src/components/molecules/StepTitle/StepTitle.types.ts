export type StepTitleProps = {
  number: number,
  title: string,
  isActive?: boolean
}
