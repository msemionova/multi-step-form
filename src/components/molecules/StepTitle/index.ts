export { default as StepTitle } from './StepTitle';
export type { StepTitleProps } from './StepTitle.types';
