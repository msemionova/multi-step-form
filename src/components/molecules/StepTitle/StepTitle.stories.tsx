import type { Meta, StoryObj } from '@storybook/react';

import StepTitle from './StepTitle';

const meta: Meta<typeof StepTitle> = {
  component: StepTitle,
};

export default meta;
type Story = StoryObj<typeof StepTitle>;

export const Default: Story = {
  args: {
    number: 1,
    title: 'Your info'
  }
};

export const Active: Story = {
  args: {
    number: 1,
    title: 'Your info',
    isActive: true
  }
};
