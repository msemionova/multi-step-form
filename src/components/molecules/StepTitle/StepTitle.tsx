import styles from './StepTitle.module.scss';
import { StepTitleProps } from './StepTitle.types';
import { CircleNumber } from 'src/components/atoms';

export default function StepTitle({ number, title, isActive = false }: StepTitleProps) {
  const label = `Step ${number}`;

  return (
    <div className={styles.step}>
      <CircleNumber isActive={isActive}>{number}</CircleNumber>
      <div>
        <span>{label}</span>
        <h2>{title}</h2>
      </div>
    </div>
  );
}
