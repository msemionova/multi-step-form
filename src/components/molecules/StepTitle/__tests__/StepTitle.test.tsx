import renderer from 'react-test-renderer';
import '@testing-library/jest-dom';

import StepTitle from '../StepTitle';

describe('<StepTitle />', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(<StepTitle number={1} title='Test'></StepTitle>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
