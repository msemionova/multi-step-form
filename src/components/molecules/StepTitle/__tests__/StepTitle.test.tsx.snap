// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<StepTitle /> renders correctly 1`] = `
<div
  className="step"
>
  <div
    className="circle"
  >
    1
  </div>
  <div>
    <span>
      Step 1
    </span>
    <h2>
      Test
    </h2>
  </div>
</div>
`;
